plutus-chain-index-core/src/Plutus/ChainIndex/DbSchema.hs

```diff
@@ -215,14 +215,42 @@ instance HasDbType ByteString where
     toDbValue = id
     fromDbValue = id
 
-deriving via ByteString instance HasDbType DatumHash
-deriving via ByteString instance HasDbType ValidatorHash
-deriving via ByteString instance HasDbType MintingPolicyHash
-deriving via ByteString instance HasDbType RedeemerHash
-deriving via ByteString instance HasDbType StakeValidatorHash
-deriving via ByteString instance HasDbType TxId
+instance HasDbType DatumHash where
```

Comment:
@kk-kien @kk-anlm Sorry I'm just curious, what's wrong with the `deriving via ByteString ...` that we need to implement these instances manually?

Reply:
From upstream, they changed from newtype BuiltinByteString to data BuiltinByteString, therefore, the original code no longer works.

Reply:
Just gave it a try, this compiled. It's significantly less verbose. Tbh, there might be a shorter way to derive `HasDbType BuiltinByteString` too.

```haskell
instance HasDbType BuiltinByteString where
    type DbType BuiltinByteString = ByteString
    toDbValue (BuiltinByteString bs) = bs
    fromDbValue = BuiltinByteString

deriving via BuiltinByteString instance HasDbType DatumHash
deriving via BuiltinByteString instance HasDbType ValidatorHash
deriving via BuiltinByteString instance HasDbType MintingPolicyHash
deriving via BuiltinByteString instance HasDbType RedeemerHash
deriving via BuiltinByteString instance HasDbType StakeValidatorHash
deriving via BuiltinByteString instance HasDbType TxId
deriving via BuiltinByteString instance HasDbType ScriptHash

deriving via ByteString instance HasDbType BlockId
```

Reply:
Made a commit here: https://github.com/MELD-labs/plutus-apps/commit/db8808cb34c44a339a52474676623c7c41b6478e

Reply:
Nice, let me try your commit. After that, I will cherry-pick your commit to the feature branch [MELD-labs:bump-plutus-v2](https://github.com/MELD-labs/plutus-apps/tree/bump-plutus-v2). Thanks for the suggestion.

Reply:
Oh, nvm, I saw your commit on the feature branch. Thanks!